### DISCUSSION TITLE
Pinia and data structures

### DISCUSSION THREAD

**Discussion by thomas-kuhlmann** [vuejs/pinia#2358] [Help and Questions]
Hi!

 In Vue 2 and Vuex there were all kind of restrictions due to the reactivity system, but Vue 3 and Pinia seem to be more flexible in that regard. 

Has anybody used data structures (i.e. via https://yomguithereal.github.io/mnemonist/ ) with Pinia? Does this cause any issues further down the road?

For example like this:

```
const useNotificationStore = defineNotifications('notifications', {
    state: () => ({ 
        notifications: new Stack() 
    }),
    actions: {
        addNotification(notification) {
            this.notifications.push(notification)
        },
        removeNotification() {
            this.notifications.pop()
        },
    }
})
```

**Comment by posva** [comment]
Custom classes will need custom serialization and reviving for SSR. Also you cannot restore the state from the devtools.
